daemon/k8s_watcher.go

```diff
@@ -980,48 +980,47 @@ func (d *Daemon) syncExternalLB(newSN, modSN, delSN *types.K8sServiceNamespace)
 func (d *Daemon) addCiliumNetworkPolicy(obj interface{}) {
 	rule, ok := obj.(*k8s.CiliumNetworkPolicy)
 	if !ok {
-		log.Warningf("Invalid third-party objected, expected CiliumNetworkPolicy, got %+v", obj)
+		log.Warningf("Received unknown object %+v, expected a CiliumNetworkPolicy object", obj)
 		return
 	}
 
-	log.Debugf("Adding k8s CRD CiliumNetworkPolicy %+v", rule)
+	log.Debugf("Adding CiliumNetworkPolicy %+v", rule)
 
 	rules, err := rule.Parse()
-	if err != nil {
-		log.Warningf("Ignoring invalid third-party policy rule: %s", err)
-		return
+	if err == nil {
+		if len(rules) > 0 {
+			_, err = d.PolicyAdd(rules, &AddOptions{Replace: true})
+		}
 	}
 
-	opts := AddOptions{Replace: true}
-	if _, err := d.PolicyAdd(rules, &opts); err != nil {
-		log.Warningf("Error while adding kubernetes network policy %+v: %s", rules, err)
-		return
+	if err != nil {
+		log.Warningf("Unable to add CiliumNetworkPolicy %s: %s", rule.Metadata.Name, err)
+	} else {
+		log.Infof("Imported CiliumNetworkPolicy %s", rule.Metadata.Name)
 	}
-
-	log.Infof("Imported third-party policy rule '%s'", rule.Metadata.Name)
 }
 
 func (d *Daemon) deleteCiliumNetworkPolicy(obj interface{}) {
 	rule, ok := obj.(*k8s.CiliumNetworkPolicy)
 	if !ok {
-		log.Warningf("Invalid third-party objected, expected CiliumNetworkPolicy, got %+v", obj)
+		log.Warningf("Received unknown object %+v, expected a CiliumNetworkPolicy object", obj)
 		return
 	}
 
-	log.Debugf("Deleting k8s CRD CiliumNetworkPolicy %+v", rule)
+	log.Debugf("Deleting CiliumNetworkPolicy %+v", rule)
 
 	rules, err := rule.Parse()
-	if err != nil {
-		log.Warningf("Ignoring invalid third-party policy rule: %s", err)
-		return
+	if err == nil {
+		if len(rules) > 0 {
+			_, err = d.PolicyDelete(rules[0].Labels)
```

Comment:
The behavior is from the time when `Specs` did not exist yet and all rules would have the same labels. I agree with the change but it will have side effects, if multiple rules contain the same labels, it will result in many warning messages in the log. I think we have to create a hashmap to extract unique labels from the rules.